orderer/common/cluster/deliver.go

```diff
@@ -40,6 +40,12 @@ type BlockPuller struct {
 	Dialer              Dialer
 	VerifyBlockSequence BlockSequenceVerifier
 	Endpoints           []EndpointCriteria
+
+	// A 'stopper' goroutine may signal the go-routine servicing PullBlock & HeightsByEndpoints to stop by closing this
+	// channel. Note: all methods of the BlockPuller must be serviced by a single goroutine, it is not thread safe.
+	// It is the responsibility of the 'stopper' not to close the channel more then once.
+	StopChannel chan struct{}
```

Comment:
The way of externally closing an internal channel is somewhat crude. I think that we should maintain the channel ourselves from within this struct, that is, to close it in a "Close" instance method.

Reply:
It is not necessarily an internal channel. It is possible to give the BlockPuller external channel, in the case of the follower, the channel that signals the follower itself to stop.